src/custom/goldfinch/index.js

```diff
@@ -0,0 +1,59 @@
+import axios from "axios";
+import {ethers} from "ethers";
+
+const provider = new ethers.providers.JsonRpcProvider(null, "mainnet");
+
+const poolTokensAddress = "0x57686612C601Cb5213b01AA8e80AfEb24BBd01df";
+const poolTokensAbi = [
+  {
+    "inputs": [],
+    "name": "totalSupply",
+    "outputs": [
+      {
+        "internalType": "uint256",
+        "name": "",
+        "type": "uint256"
+      }
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  }
+];
+const poolTokensContract = new ethers.Contract(poolTokensAddress, poolTokensAbi, provider);
+
+const metadataBaseURI = "https://us-central1-goldfinch-frontends-prod.cloudfunctions.net/poolTokenMetadata";
+
+export const fetchToken = async (_chainId, { contract, tokenId }) => {
+  const response = await axios.get(`${metadataBaseURI}/${tokenId}`);
+  return {
+    contract,
+    tokenId,
+    name: response.data.name,
+    imageUrl: response.data.image,
+    attributes: response.data.attributes,
+  };
+};
+
+export const fetchContractTokens = async (_chainId, contract, continuation) => {
```

Comment:
Currently broken due to `metadata` key not existing. See question below. Can remove this function if not needed